diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt

```diff
@@ -279,9 +280,21 @@ class IdentifierNaming : Rule("identifier-naming") {
                                       isVariable: Boolean) {
         nodes.forEach {
             if (!(it.checkLength(MIN_IDENTIFIER_LENGTH..MAX_IDENTIFIER_LENGTH) ||
-                            (ONE_CHAR_IDENTIFIERS.contains(it.text)) && isVariable)) {
+                            ONE_CHAR_IDENTIFIERS.contains(it.text) && isVariable || validCatchIdentifier(it)
+
+                            )) {
                 IDENTIFIER_LENGTH.warn(configRules, emitWarn, isFixMode, it.text, it.startOffset)
             }
         }
     }
+
+    /**
+     * exception case for identifiers used in catch block:
+     * catch (e: Exception) {}
+     */
+    private fun validCatchIdentifier(node: ASTNode): Boolean {
+        return node.text == "e" &&
+                node.findParentNodeWithSpecificType(CATCH) != null &&
+                node.findParentNodeWithSpecificType(VALUE_PARAMETER_LIST) != null
```

Comment:
Won't this condition also trigger on, e.g., parameters of functions declared inside catch block? I think checking type of parents with explicit index would be more reliable.

Reply:
you think that in `catch ()` block there can be something except the `e: Exception`? 
Could you please provide example just for academic interest?

Reply:
No, I mean something like the following:
```
CATCH: "catch (e: Exception) {
        fun foo(e: Int) { } 
    }"
- catch: "catch"
- WHITE_SPACE: " "
- VALUE_PARAMETER_LIST: "(e: Exception)"
- WHITE_SPACE: " "
- BLOCK: "{
        fun foo(e: Int) { } 
    }"
```
What I was trying to say is that checking if node has parents with these types isn't enough to prove that it is node from catch condition and not body.

Reply:
I will check prev keyword then